compiler/src/iree/compiler/Dialect/HAL/IR/HALOps.cpp

```diff
@@ -454,24 +454,47 @@ static void printWorkgroupCountRegion(OpAsmPrinter &p, Operation *op,
 }
 
 //===----------------------------------------------------------------------===//
-// custom<OptionalWorkgroupCountRegion>($body)
+// custom<ExportConditionRegion>($body)
 //===----------------------------------------------------------------------===//
 
-static ParseResult parseOptionalWorkgroupCountRegion(OpAsmParser &parser,
-                                                     Region &body) {
-  // Empty region if not specified.
-  if (failed(parser.parseOptionalKeyword("count"))) {
-    return success();
+static ParseResult parseExportConditionRegion(OpAsmParser &parser,
+                                              Region &body) {
+  SmallVector<OpAsmParser::Argument> args;
+  if (failed(parser.parseArgumentList(args, AsmParser::Delimiter::Paren,
+                                      /*allowType=*/true,
+                                      /*allowAttrs=*/true))) {
+    return failure();
+  }
+
+  // Return types must be an i1.
+  SmallVector<Type> returnTypes;
+  if (failed(parser.parseArrowTypeList(returnTypes))) {
+    return failure();
+  }
+  if (returnTypes.size() != 1 ||
+      !llvm::all_of(returnTypes, [](Type type) { return type.isInteger(1); })) {
```

Comment:
Do we care about signlessness? The printer looks like it produces a signless result.
```suggestion
      !llvm::all_of(returnTypes, [](Type type) { return type.isSignlessInteger(1); })) {
```

Reply:
it's a boolean, so don't care :P `isInteger(1)` is the common way of doing this so I'm going to stick with that. An `isBool()` would be a useful bit of syntactic sugar.